### PR TITLE
Add RBAC policy parser methods

### PR DESCRIPTION
**Pull request by bernardo-bastos**
Adds the RBAC policy parser methods.

Rev: DSN-CV2ND9

### CODE REVIEW

File: pkg/auth/rbac.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Inits the RBAC Config struct
+func initRbacConfig() RBACconfig {
+	return RBACconfig{
+		allowedApp:     []string{"EnvironmentLock", "EnvironmentApplicationLock", "Deploy", "Undeploy", "EnvironmentFromApp"},
```
Comment (hannesg) [inline pkg/auth/rbac.go:11]: Please make the name `EnvironmentFromApp` consistent and use `EnvironmentFromApplication`

File: pkg/auth/rbac.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	// Validate the permission action
+	err = cfg.validateAction(c[3])
+	if err != nil {
+		return nil, err
+	}
+	return &permission{
+		Role:        c[1],
+		Application: c[2],
+		Action:      c[3],
+		Environment: c[4],
```
Comment (hannesg) [inline pkg/auth/rbac.go:74]: You are not validating the environment name. Probably you should.

Comment (bernardo-bastos) [reply pkg/auth/rbac.go:97]: Yes, good ideia. I added a check for environments and a unit test for it as well.

File: pkg/auth/rbac.go (hunk 1 above)
Comment (bernardo-bastos) [reply pkg/auth/rbac.go]: Fixed it.

**Review by hannesg — APPROVED** [review]

File: services/cd-service/pkg/cmd/server.go [hunk 3]
```diff
@@ -110,7 +130,7 @@ func RunServer() {
 		if err != nil {
 			logger.FromContext(ctx).Fatal("dex.read.error", zap.Error(err))
```
[the remark this answers is not in the document]
Comment (bernardo-bastos) [reply services/cd-service/pkg/cmd/server.go]: True, I restructured the if statement not to do that.

File: pkg/auth/rbac.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+func InitRbacConfig() RBACconfig {
+	return RBACconfig{
+		allowedApp:     []string{"EnvironmentLock", "EnvironmentApplicationLock", "Deploy", "Undeploy", "EnvironmentFromApplication"},
+		allowedActions: []string{"Create", "Delete"},
+	}
+}
+
+// Stores the RBAC allowed Applications and Actions
+type RBACconfig struct {
+	allowedApp     []string
```
Comment (sven-urbanski-freiheit-com) [inline pkg/auth/rbac.go:36]: please rename to `allowedApps`

File: pkg/auth/rbac.go [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+	}
+	for _, a := range c.allowedActions {
+		if a == action {
+			return nil
+		}
+	}
+	return fmt.Errorf("invalid action %s", action)
+}
+
+func (c *RBACconfig) validateEnvs(env string) error {
```
Comment (sven-urbanski-freiheit-com) [inline pkg/auth/rbac.go:64]: please rename env -> envs

**Pull request merged**